pysoa/server/standalone.py

```diff
@@ -19,14 +22,63 @@
     exit(99)
 
 
+def _get_arg_parser():
+    import argparse
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '-f', '--fork-processes', '--fork',
+        help='The number of processes to fork (if 0, 1, or none, no process is forked; the server is run directly)',
+        required=False,
+        type=int,
+        default=0,
+    )
+    return parser
+
+
+def _get_args(parser):
+    return parser.parse_known_args()[0]
+
+
+def _run_server(args, server_class):
+    if args.fork_processes > 1:
+        import multiprocessing
+        import signal
+        import time
+
+        processes = []
+        cmd = {'keep_starting': True}
+
+        def _stop(*_):
+            cmd['keep_starting'] = False
+
+        signal.signal(signal.SIGINT, _stop)
+        signal.signal(signal.SIGTERM, _stop)
+
+        for i in range(0, args.fork_processes):
+            process = multiprocessing.Process(target=server_class.main, name='pysoa-worker-{}'.format(i))
+
+            if not cmd['keep_starting']:
+                # Got shutdown signal while starting up, so don't keep starting up
+                break
+
+            processes.append(process)
+            process.start()
+            time.sleep(0.3)
+
+        for process in processes:
+            process.join()
```

Comment:
But how would the sub-processes know to join, if the signals to terminate are not passed to them?

Reply:
Yep!